src/main/java/org/jabref/logic/pdf/PdfAnnotationImporter.java

```diff
@@ -39,115 +44,123 @@
+        PDFTextStripperByArea stripperByArea = new PDFTextStripperByArea();
+        COSArray quadsArray = (COSArray) annotation.getDictionary().getDictionaryObject(COSName.getPDFName("QuadPoints"));
+        String highlightedText = "";
+        for (int j = 1,
+             k = 0;
+             j <= (quadsArray.size() / 8);
+             j++) {
+
+            COSFloat upperLeftX = (COSFloat) quadsArray.get(k);
+            COSFloat upperLeftY = (COSFloat) quadsArray.get(1 + k);
+            COSFloat upperRightX = (COSFloat) quadsArray.get(2 + k);
+            COSFloat upperRightY = (COSFloat) quadsArray.get(3 + k);
+            COSFloat lowerLeftX = (COSFloat) quadsArray.get(4 + k);
+            COSFloat lowerLeftY = (COSFloat) quadsArray.get(5 + k);
+
+            k += 8;
+
+            float ulx = upperLeftX.floatValue() - 1;
+            float uly = upperLeftY.floatValue();
+            float width = upperRightX.floatValue() - lowerLeftX.floatValue();
+            float height = upperRightY.floatValue() - lowerLeftY.floatValue();
+
+            PDRectangle pageSize = page.getMediaBox();
+            uly = pageSize.getHeight() - uly;
+
+            Rectangle2D.Float rectangle = new Rectangle2D.Float(ulx, uly, width, height);
+            stripperByArea.addRegion("highlightedRegion", rectangle);
+            stripperByArea.extractRegions(page);
+            String highlightedTextInLine = stripperByArea.getTextForRegion("highlightedRegion");
+
+            if (j > 1) {
+                highlightedText = highlightedText.concat(highlightedTextInLine);
+            } else {
+                highlightedText = highlightedTextInLine;
             }
         }
-        try {
-            document.close();
-        } catch (IOException ignored) {
+
+        if (highlightedText.trim().isEmpty()) {
+            highlightedText = "JabRef: The highlighted area does not contain any legible text!";
         }
-        return annotationsList;
+
+        return highlightedText;
     }
 
-    /**
-     *
-     * @param path the absolute path to the pdf file
-     * @return a PDDocument representing the pdf file
-     * @throws IOException if the document cannot be read from path
-     */
-    private PDDocument importPdfFile(final String path) throws IOException {
+    private Optional<Path> validatePath(Path path, BibDatabaseContext context) {
+        Objects.requireNonNull(path);
+
+        if (!path.toString().toLowerCase(Locale.ROOT).endsWith(".pdf")) {
+            LOGGER.warn(String.format("File %s does not end with .pdf!", path.toString()));
+            return Optional.empty();
+        }
 
-            if(path.toLowerCase(Locale.ROOT).endsWith(".pdf")){
-               return PDDocument.load("/"+ path);
+        if (!Files.exists(path)) {
+            Optional<File> importedFile = FileUtil.expandFilename(context, path.toString(),
+                    JabRefPreferences.getInstance().getFileDirectoryPreferences());
```

Comment:
Yes this change would propagate to EntryAnnotationImporter and FileAnnotationCache (and probably further places) until one ends up in gui. It's not that important, you can also leave it as it is right now.